lib/utils/addr.go

```diff
@@ -247,3 +248,61 @@ func IsLoopback(host string) bool {
 	}
 	return false
 }
+
+// GuessIP tries to guess an IP address this machine is reachable at on the
+// internal network, always picking IPv4 from the internal address space
+//
+// If no internal IPs are found, it returns 127.0.0.1 but it never returns
+// an address from the public IP space
+func GuessHostIP() (ip net.IP, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// collect the list of all IPv4s
+	ips := make([]net.IP, 0)
+	for _, iface := range ifaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			log.Warn(err)
+			continue
+		}
+		for _, addr := range addrs {
+			a, ok := addr.(*net.IPAddr)
+			if ok {
+				ip = a.IP
+			} else {
+				in, ok := addr.(*net.IPNet)
+				if ok {
+					ip = in.IP
+				} else {
+					continue
+				}
+			}
+			if ip.To4() == nil || ip.IsLoopback() || ip.IsMulticast() {
+				continue
+			}
+			ips = append(ips, ip)
+		}
+	}
+	for i := range ips {
```

Comment:
btw, for vagrant case this prioritization will do the opposite in many cases - the default network for `Vagrant` is `172.xx` while the bridge IP is `10.x` so in Vagrant all nodes in the cluster will advertise the wrong IP